merlin/status/status.go

```diff
@@ -0,0 +1,79 @@
+/*
+Package status provides an updater for a Merlin frontend to update ingress statuses.
+*/
+package status
+
+import (
+	"errors"
+
+	"github.com/sky-uk/feed/controller"
+	"github.com/sky-uk/feed/k8s"
+	"github.com/sky-uk/feed/util"
+	"k8s.io/client-go/pkg/api/v1"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	internalLabelValue = "internal"
+	externalLabelValue = "internet-facing"
+)
+
+// Config for creating a new Merlin status updater.
+type Config struct {
+	InternalHostname string
+	ExternalHostname string
+	KubernetesClient k8s.Client
+}
+
+// New creates a new Merlin frontend status updater.
+func New(conf Config) (controller.Updater, error) {
+	loadBalancers := make(map[string][]v1.LoadBalancerIngress)
+	loadBalancers[internalLabelValue] = util.GenerateLoadBalancerStatus([]string{conf.InternalHostname})
+	loadBalancers[externalLabelValue] = util.GenerateLoadBalancerStatus([]string{conf.ExternalHostname})
```

Comment:
The concept of "internal" and "external" ingress only makes sense for AWS. All our ingresses will be accessible via the external and internal VIPs, at least in the current setup.

Reply:
Addressed with new `merlinInternetFacingVIP` flag which allows setting the external ip address (currently just used as metadata for setting the status). At a later date we may want to query Merlin for both this VIP and the `merlinVIP`.